src/jobs/models/manager.ts

```diff
@@ -138,34 +161,129 @@ export class JobManager {
+        job_id: jobId,
+      },
+    });
+
+    const deleteJob = this.prisma.job.delete({
+      where: {
+        id: jobId,
+      },
+    });
+
+    try {
+      await this.prisma.$transaction([deleteStages, deleteJob]);
+    } catch (err) {
+      if (err instanceof Prisma.PrismaClientKnownRequestError && err.code === prismaKnownErrors.recordNotFound) {
+        throw new JobNotFoundError(JOB_NOT_FOUND_MSG);
+      }
+
+      throw err;
+    }
+  }
+
+  public async addStages(jobId: string, stagesPayload: StageCreateModel[]): Promise<JobModel> {
+    // todo - use stages machine on next phase when will be implemented
+    const createJobActor = createActor(jobStateMachine).start();
+    const persistenceSnapshot = createJobActor.getPersistedSnapshot();
+
+    const queryExistJobBody = {
+      where: {
+        id: jobId,
+      },
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      select: { data: true },
+    };
+
+    const jobBeforeUpdate = await this.prisma.job.findUnique(queryExistJobBody);
+
+    if (!jobBeforeUpdate) {
+      throw new JobNotFoundError(JOB_NOT_FOUND_MSG);
+    }
+
+    const unifiedStagesInput = jobBeforeUpdate.data.stages ? jobBeforeUpdate.data.stages.concat(stagesPayload) : stagesPayload;
+
+    const queryBody = {
+      where: {
+        id: jobId,
+      },
+      data: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        Stage: {
+          create: stagesPayload.map((stageData) => ({
+            data: stageData.data,
+            name: stageData.type,
+            xstate: persistenceSnapshot,
+            userMetadata: stageData.userMetadata,
+            status: StageOperationStatus.CREATED,
+          })),
+        },
+        data: {
+          ...jobBeforeUpdate.data,
+          stages: unifiedStagesInput,
+        },
+      },
+
+      include: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        Stage: true, // Include the created stages in the return value
+      },
+    };
+    try {
+      const job = await this.prisma.job.update(queryBody);
+
+      return this.convertPrismaToJobResponse(job);
+    } catch (error) {
+      this.logger.error(`Failed adding stage to job with error: ${(error as Error).message}`);
+
+      throw error;
+    }
+  }
+
   /**
    * This method is used to get a job entity by its id from the database.
    * @param jobId unique identifier of the job.
    * @returns The job entity if found, otherwise null.
    */
-  public async getJobEntityById(jobId: string): Promise<Prisma.JobGetPayload<Record<string, never>> | null> {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  public async getJobEntityById(jobId: string): Promise<Prisma.JobGetPayload<{ include: { Stage: true } }> | null> {
     const queryBody = {
       where: {
         id: jobId,
       },
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      include: { Stage: true },
```

Comment:
same as before, im not sure i want all the stages all the time.

Reply:
done